lib/spack/spack/cmd/fetch.py

```diff
@@ -16,24 +17,53 @@
 
 
 def setup_parser(subparser):
-    arguments.add_common_arguments(subparser, ['no_checksum'])
+    arguments.add_common_arguments(subparser, ["no_checksum"])
     subparser.add_argument(
-        '-m', '--missing', action='store_true',
-        help="fetch only missing (not yet installed) dependencies")
+        "-m",
+        "--missing",
+        action="store_true",
+        help="fetch only missing (not yet installed) dependencies",
+    )
     subparser.add_argument(
-        '-D', '--dependencies', action='store_true',
-        help="also fetch all dependencies")
-    arguments.add_common_arguments(subparser, ['specs'])
+        "-D",
+        "--dependencies",
+        action="store_true",
+        help="also fetch all dependencies",
+    )
+    arguments.add_common_arguments(subparser, ["specs"])
+    subparser.epilog = (
+        "With an active environment, the specs "
+        "parameter can be omitted. In this case all (uninstalled"
+        ", in case of --missing) specs from the environment are fetched"
+    )
 
 
 def fetch(parser, args):
-    if not args.specs:
-        tty.die("fetch requires at least one package argument")
+    if args.specs:
+        specs = spack.cmd.parse_specs(args.specs, concretize=True)
+    else:
+        # No specs were given explicitly, check if we are in an
+        # environment. If yes, check the missing argument, if yes
+        # fetch all uninstalled specs from it otherwise fetch all.
+        # If we are also not in an environment, complain to the
+        # user that we don't know what to do.
+        env = ev.get_env(args, "fetch")
+        if env:
+            if args.missing:
+                specs = env.uninstalled_specs()
+            else:
+                specs = env.all_specs()
+            if specs == []:
+                tty.die(
```

Comment:
I would think this PR should concretize the environment, the same way that the `spack fetch` command currently concretizes specs.